tests/unit/lms/services/lti_h_test.py

```diff
@@ -1,42 +1,59 @@
 from unittest import mock
+from unittest.mock import create_autospec
 
 import pytest
 from pyramid.httpexceptions import HTTPBadRequest, HTTPInternalServerError
 
 from lms.models import GroupInfo
 from lms.services import HAPIError, HAPINotFoundError
-from lms.services.lti_h import LTIHService
+from lms.services.lti_h import Group, LTIHService
 from lms.values import HUser, LTIUser
 
 
-class TestUpsertCourseGroup:
-    def test_it_does_nothing_if_the_feature_isnt_enabled(
-        self, context, h_api, lti_h_svc
+class TestSync:
+    def test_single_group_sync_calls_sync(self, lti_h_svc, group):
+        lti_h_svc.sync = create_autospec(lti_h_svc.sync)
+
+        lti_h_svc.single_group_sync()
+
+        lti_h_svc.sync.assert_called_once_with(groups=[group])
+
+    def test_sync_does_nothing_if_provisioning_is_disabled(
+        self, context, lti_h_svc, h_api, group
     ):
-        # If the auto provisioning feature isn't enabled for this application
-        # instance then upsert_course_group() doesn't do anything - just calls the
-        # wrapped view.
         context.provisioning_enabled = False
 
-        lti_h_svc.upsert_course_group()
+        lti_h_svc.sync(groups=[group])
+
+        h_api.upsert_user.assert_not_called()
+
+    def test_sync_catches_HAPIErrors(self, h_api, lti_h_svc, group):
+        h_api.upsert_user.side_effect = HAPIError
+
+        with pytest.raises(HTTPInternalServerError):
+            lti_h_svc.sync(groups=[group])
+
+    @pytest.fixture
+    def group(self):
+        return Group(name="test_group_name", groupid="test_groupid")
 
-        h_api.update_group.assert_not_called()
 
```

Comment:
Below here all the tests are as before. I've just changed the previous individual method tests to be called from the top level. These will all be gone soon.